giraph-core/src/main/java/org/apache/giraph/conf/GiraphConstants.java

```diff
@@ -1238,5 +1239,27 @@
   BooleanConfOption PREFER_IP_ADDRESSES =
       new BooleanConfOption("giraph.preferIP", false,
       "Prefer IP addresses instead of host names");
+
+  /**
+   * Timeout for "waitForever", when we need to wait for zookeeper.
+   * Since we should never really have to wait forever.
+   * We should only wait some reasonable but large amount of time.
+   */
+  LongConfOption WAIT_FOREVER_ZOOKEEPER_TIMEOUT_MSEC =
```

Comment:
Nit: since we are not waiting forever anymore, I'd drop word forever from everywhere (forever and timeout have opposite meaning :-))